### PR TITLE
Merge openshift-tuned

### PR DESCRIPTION
**Pull request by sjug**
Added assets and pkg from openshift-tuned repo, calling wholesale from main based on symbolic link name. This pass has the minimal changes required to function.

### CODE REVIEW

File: pkg/tuned/tuned.go [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+)
+
+// Global variables
+var (
+	done               = make(chan bool, 1)
+	tunedExit          = make(chan bool, 1)
+	terminationSignals = []os.Signal{syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT}
+	version            string // programName version
+	cmd                *exec.Cmd
+	// Flags
```
Comment (jmencak) [inline pkg/tuned/tuned.go]: Let's remove these 2 lines.

File: cmd/cluster-node-tuning-operator/main.go [hunk 2]
```diff
@@ -25,24 +27,36 @@ func printVersion() {
 }
 
 func main() {
-	klog.InitFlags(nil)
-	flag.Parse()
+	operatorExec := regexp.MustCompile(`.*cluster-node-tuning-operator$`)
```
Comment (jmencak) [inline cmd/cluster-node-tuning-operator/main.go]: Do we need to use regexp for this?  Please consider "path/filepath" Base().

File: cmd/cluster-node-tuning-operator/main.go [hunk 3]
```diff
@@ -25,24 +27,36 @@ func printVersion() {
...
+		}
+
+		err = controller.Run(stopCh)
+		if err != nil {
+			klog.Fatalf("error running controller: %s", err.Error())
+		}
+	case operandExec.MatchString(runAs):
+		tuned.Run(boolVersion)
+	default:
+		klog.Fatalln("Application should be run as \"cluster-node-tuning-operator\" or \"openshift-tuned\"")
```
Comment (jmencak) [inline cmd/cluster-node-tuning-operator/main.go]: Extra newline?

Comment (sjug) [reply cmd/cluster-node-tuning-operator/main.go]: Removed

File: cmd/cluster-node-tuning-operator/main.go (hunk 2 above)
Comment (sjug) [reply cmd/cluster-node-tuning-operator/main.go]: Good call

File: pkg/tuned/tuned.go (hunk 1 above)
Comment (sjug) [reply pkg/tuned/tuned.go]: Done

**Comment by sjug** [comment]
PR'ed some fixes as you mentioned Jiri, where should we specify the changed entry points for the image?

**Comment by jmencak** [comment]
Also, could also the references to the operand's repo in README.md `https://github.com/openshift/openshift-tuned` be removed?

File: cmd/cluster-node-tuning-operator/main.go [hunk 4]
```diff
@@ -25,24 +27,38 @@ func printVersion() {
...
-	if err != nil {
-		klog.Fatalf("error running controller: %s", err.Error())
+		err = controller.Run(stopCh)
+		if err != nil {
+			klog.Fatalf("error running controller: %s", err.Error())
+		}
+	case operandFilename:
+		tuned.Run(boolVersion)
+	default:
+		klog.Fatalf("Application should be run as \"%s\" or \"%s\"", operatorFilename, operandFilename)
```
[the remark this answers is not in the document]
Comment (sjug) [reply cmd/cluster-node-tuning-operator/main.go]: Done

File: Dockerfile [hunk 5]
```diff
@@ -3,12 +3,30 @@ WORKDIR /go/src/github.com/openshift/cluster-node-tuning-operator
...
+    ARCH_DEP_PKGS=$(if [ "$(uname -m)" != "s390x" ]; then echo -n hdparm kernel-tools ; fi) && \
+    yum install --setopt=tsflags=nodocs -y $INSTALL_PKGS $ARCH_DEP_PKGS && \
+    rpm -V $INSTALL_PKGS $ARCH_DEP_PKGS && \
+    (LC_COLLATE=C cat patches/*.diff | patch -Np1 -d /usr/lib/python*/site-packages/tuned/ || :) && \
+    touch /etc/sysctl.conf && \
+    yum -y remove patch && \
+    yum clean all && \
+    rm -rf /var/cache/yum ~/patches && \
+    useradd cluster-node-tuning-operator
 USER cluster-node-tuning-operator
```
[the remark this answers is not in the document]
Comment (sjug) [reply Dockerfile]: Will try as UID below

File: Dockerfile [hunk 6]
```diff
@@ -3,12 +3,30 @@ WORKDIR /go/src/github.com/openshift/cluster-node-tuning-operator
...
+      " && \
+    ARCH_DEP_PKGS=$(if [ "$(uname -m)" != "s390x" ]; then echo -n hdparm kernel-tools ; fi) && \
+    yum install --setopt=tsflags=nodocs -y $INSTALL_PKGS $ARCH_DEP_PKGS && \
+    rpm -V $INSTALL_PKGS $ARCH_DEP_PKGS && \
+    (LC_COLLATE=C cat patches/*.diff | patch -Np1 -d /usr/lib/python*/site-packages/tuned/ || :) && \
+    touch /etc/sysctl.conf && \
+    yum -y remove patch && \
+    yum clean all && \
+    rm -rf /var/cache/yum ~/patches && \
+    useradd cluster-node-tuning-operator
```
[the remark this answers is not in the document]
Comment (sjug) [reply Dockerfile]: Done

File: Makefile [hunk 7]
```diff
@@ -28,6 +28,7 @@ all: build
 
 build: $(BINDATA) pkg/generated
 	$(GO_BUILD_RECIPE)
+	ln -s $(PACKAGE_BIN) $(OUT_DIR)/openshift-tuned
```
[the remark this answers is not in the document]
Comment (sjug) [reply Makefile]: Went with `-sf`

**Comment by jmencak** [comment]
After adding:
```
diff --git a/manifests/04-operator.yaml b/manifests/04-operator.yaml
index 53a0137..3fe41a2 100644
--- a/manifests/04-operator.yaml
+++ b/manifests/04-operator.yaml
@@ -13,6 +13,9 @@ spec:
       labels:
         name: cluster-node-tuning-operator
     spec:
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 499
       serviceAccountName: cluster-node-tuning-operator
       nodeSelector:
         node-role.kubernetes.io/master: ""
```
we should be good to go to squash and merge.

**Comment by jmencak** [comment]
This looks good now, can we squash this?

**Pull request merged**